ui/appui-react/src/appui-react/content/FloatingViewportContent.tsx

```diff
@@ -27,8 +27,10 @@ export interface FloatingViewportContentProps {
   onContextMenu?: (e: React.MouseEvent) => boolean;
   /** viewport/content control uniqueId */
   contentId: string;
-  /** The initial view state used to create the viewport */
-  initialViewState: ViewState;
+  /** The initial view state used to create the viewport, or a function that returns it (will refresh when the function changes) */
+  initialViewState: ViewStateProp;
+  /** Function to get a reference to the ScreenViewport */
+  viewportRef?: React.ForwardedRef<ScreenViewport>;
```

Comment:
I think we should keep this as `React.Ref<ScreenViewport>`, to merge the `viewportRef` w/ `onViewportRef` you could use `useRefs` or `mergeRefs` from `core-react`..

Reply:
Ref are immutable, so we cant assing to it, so I suggested the ForwardedRef...

Reply:
After looking around a bit, I think this type is indeed not meant to be used like that, so my suggestion would be wrong, That being said, React "ref" are not meant to be props and are handled differently, so I guess we might as well just use Gerardas suggestion (or revert back to what it was before, as these are not real refs, the goal being to have the same behavior as ViewportComponent ?) @GerardasB  ?

Reply:
Yes, I guess we can match what's currently in the `ViewportComponent`. We can refactor both components later.